### PR TITLE
PodmanPy Containers and system areas of concern

### PR DESCRIPTION
**Pull request by jwhonce**
PodmanPy Containers and system areas of concern

    * Implement most containers and system calls
    * Pylint run on all new code
    * Updates to code and unittests

Signed-off-by: Jhon Honce <jhonce@redhat.com>

### CODE REVIEW

**Comment by baude** [comment]
lgtm

File: podman/domain/containers.py [hunk 1]
```diff
@@ -63,33 +70,64 @@ def attach(self, **kwargs) -> Union[str, Iterator[str]]:
...
-            author (str): Name of commit author.
-            changes (str): Instructions to apply during commit.
-            conf (Dict[str, Any]): Configuration for the container.
+            author (str): Name of commit author
+            changes (List[str]): Instructions to apply during commit
+            comment (List[str]): instructions to apply while committing in Dockerfile format
+            conf (Dict[str, Any]): ignored
+            format (str): format of the image manifest and metadata
+            message (str): Commit message to include with Image
+            pause (bool): pause the container before committing it
```
Comment (TomSweeneyRedHat) [inline podman/domain/containers.py:93]: Nit for consideration.  Some of the words have the first letter uppercased above after the semi-colon, others are not.  I'd prefer all to be uppercased, but more so, consistent.

File: podman/domain/containers_manager.py [hunk 2]
```diff
@@ -0,0 +1,361 @@
...
+                For example,
+                    /dev/sda:/dev/xvda:rwm allows the container to have read-write access to the
+                    host's /dev/sda via a node named /dev/xvda inside the container.
+            dns (List[str]): Set custom DNS servers.
+            dns_opt (List[str]): Additional options to be added to the container's resolv.conf file.
+            dns_search (List[str]): DNS search domains.
+            domainname (Union[str, List[str]]): Set custom DNS search domains.
+            entrypoint (Union[str, List[str]]): The entrypoint for the container.
+            environment (Union[Dict[str, str], List[str]): Environment variables to set inside
+                the container, as a dictionary or a List[str] in the format ["SOMEVARIABLE=xxx"].
```
Comment (TomSweeneyRedHat) [inline podman/domain/containers_manager.py:83]: might be nice to show two variables here.

**Comment by TomSweeneyRedHat** [comment]
A couple of nits that can be ignored or updated later if you wish.
LGTM

**Comment by TomSweeneyRedHat** [comment]
LGTM

**Pull request merged**